pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -103,6 +116,48 @@ func GetDockercfgFile(path string) string {
 	return cfgPath
 }
 
+func readSpecificDockerConfigJSONFile(filePath string) bool {
+	var contents []byte
+	var err error
+
+	if contents, err = ioutil.ReadFile(filePath); err != nil {
+		log.V(4).Infof("error reading file: %v", err)
+		return false
+	}
+	return readDockerConfigJSONFileFromBytes(contents)
+}
+
+func readDockerConfigJSONFileFromBytes(contents []byte) bool {
+	var cfgJSON credentialprovider.DockerConfigJson
+	if err := json.Unmarshal(contents, &cfgJSON); err != nil {
+		log.V(4).Infof("while trying to parse blob %q: %v", contents, err)
+		return false
+	}
+	return true
+}
+
+// GetDockerConfigPath returns the first path that provides a valid DockerConfig;
+// modified elements from credentialprovider methods called from GetDockerConfig, following the same order of precedenced
+// articulated in GetDockerConfig, via the order of file names set in dockerFilesToExamine
+func GetDockerConfigPath(paths []string) string {
+	for _, configPath := range paths {
+		for _, file := range dockerFilesToExamine {
+			absDockerConfigFileLocation, err := filepath.Abs(filepath.Join(configPath, file))
```

Comment:
`GetDockerConfig` == `credentialprovider.ReadDockerConfigJSONFile` only tries `configJsonFileName = "config.json"`, not the other two, AFAICS, so I’m not sure why all three need to be checked here.  It doesn’t quite _hurt_, of course.

Reply:
If you look at https://github.com/openshift/builder/blob/master/pkg/build/builder/cmd/dockercfg/cfg.go#L107-L114

```
func GetDockerConfig(path []string) (cfg credentialprovider.DockerConfig, err error) {
	if cfg, err = credentialprovider.ReadDockerConfigJSONFile(path); err != nil {
		if cfg, err = ReadDockerConfigJsonFileGeneratedFromSecret(path); err != nil {
			cfg, err = credentialprovider.ReadDockercfgFile(path)
		}
	}
	return cfg, err
}
```

Which is what we *used* to call, `credentialprovider.ReadDockerConfigJSONFile(path)` checks for `config.json` at https://github.com/openshift/builder/blob/master/vendor/k8s.io/kubernetes/pkg/credentialprovider/config.go#L121-L144

`ReadDockerConfigJsonFileGeneratedFromSecret(path)` calls `credentialprovider.ReadSpecificDockerConfigJsonFile(filepath.Join(filePath, DockerConfigJsonKey))` where that is `DockerConfigJsonKey` is `.dockerconfigjson`

and the final `credentialprovider.ReadDockercfgFile(path)` at https://github.com/openshift/builder/blob/master/vendor/k8s.io/kubernetes/pkg/credentialprovider/config.go#L90-L117 looks for `.dockercfg`

I'm following that flow, but rather than getting a `DockerConfig`, I'm getting the fully qualified path of the file found.  We ultimately set that in `AuthFilePath`

Reply:
You’re completely right, I don’t know how I missed the two other cases.

Reply:
no worries - good we are on the same page